### PR TITLE
Parse Json and add activity indicator

### PR DESCRIPTION
**Pull request by agustingodoyc**
**STORY**
Parse Json and add activity indicator
DESCRIPTION
Parse Json on background and add activity indicator
EVIDENCE
https://user-images.githubusercontent.com/109157659/186528073-22de7fec-1db7-4695-8508-cecea7554958.mp4

### CODE REVIEW

File: Ted Talk/Ted Talk/Model/Parser.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+public class Parser {
+    
+    enum ParseError: Error {
+        case fileNotFound
+        case decodeError
+    }
+    
+    func parse(_ file: String, completionHandler: @escaping (Result<[TedTalk], ParseError>) -> Void){
+        DispatchQueue.global(qos: .background).async {
+            if let fileLocation = Bundle.main.url(forResource: file, withExtension: "json") {
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Model/Parser.swift]: do not use if let, use guard statement instead.

File: Ted Talk/Ted Talk/Model/Parser.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                    }
+                } catch {
+                    DispatchQueue.main.async {
+                        completionHandler(.failure(.decodeError))
+                    }
+                }
+            } else {
+                DispatchQueue.main.async {
+                    completionHandler(.failure(.fileNotFound))
+                }
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Model/Parser.swift]: this is repetitive and should not be responsibility of the parse switch to main thread

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]

**Comment by ezequielvelezglobant** [comment]
Also you need a Module_2/main branch and this PR should point to that branch instead of main.

**Comment by ezequielvelezglobant** [comment]
Add some Unit Test

File: Ted Talk/Ted Talk/Model/Parser.swift (hunk 2 above)
Comment (agustingodoyc) [reply Ted Talk/Ted Talk/Model/Parser.swift]: https://github.com/agustingodoyc/iOS_Academy_base_project/pull/1/commits#:~:text=Aug%2025%2C%202022-,Feature%201%20fixes%201,-agustingodoyc%20committed%2020

Comment (agustingodoyc) [reply Ted Talk/Ted Talk/Model/Parser.swift]: https://github.com/agustingodoyc/iOS_Academy_base_project/pull/1/commits/082cd50052078ed54c57198d523dca6f466dec62

File: Ted Talk/Ted Talk/Model/Parser.swift (hunk 1 above)
Comment (agustingodoyc) [reply Ted Talk/Ted Talk/Model/Parser.swift]: https://github.com/agustingodoyc/iOS_Academy_base_project/pull/1/commits/b91a34cb2a6dd7e1527efeb5bec967bbbbcde4ea

File: Ted Talk/Ted Talk/Controller/ViewController.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    
+    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        activityIndicator.hidesWhenStopped = true
+        self.activityIndicator.startAnimating()
+        DataManager().getTalks() { talks in
+            self.data = talks
+            self.activityIndicator.stopAnimating()
```
Comment (JMVarelaGlobant) [inline Ted Talk/Ted Talk/Controller/ViewController.swift]: It's missing a delay to show the spinner for more time

**Review by JMVarelaGlobant — CHANGES_REQUESTED** [review]

File: Ted Talk/Ted Talk/Model/DataManager.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import Foundation
+
+public class DataManager {
+    
+    private var file: String
+    
+    init (file: String? = "tedTalks") {
+        self.file = file!
+    }
```
Comment (ezequielvelezglobant) [inline Ted Talk/Ted Talk/Model/DataManager.swift]: if you set a default value, you don't need the `?`.
Also remember never use `!`

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]

File: Ted Talk/Ted Talk/Controller/ViewController.swift (hunk 3 above)
Comment (agustingodoyc) [reply Ted Talk/Ted Talk/Controller/ViewController.swift]: [cfc4e52](https://github.com/agustingodoyc/iOS_Academy_base_project/pull/1/commits/cfc4e52c3ed6b01ae61c5c8bd5f8ad356fe93fa5)

File: Ted Talk/Ted Talk/Model/DataManager.swift (hunk 4 above)
Comment (agustingodoyc) [reply Ted Talk/Ted Talk/Model/DataManager.swift]: [46cf05c](https://github.com/agustingodoyc/iOS_Academy_base_project/pull/1/commits/46cf05c6987bf371d5f39861850b2577daf0d291)

**Review by JMVarelaGlobant — APPROVED** [review]

**Review by ezequielvelezglobant — APPROVED** [review]

**Pull request merged**